rust_snuba/src/types.rs

```diff
@@ -70,28 +92,28 @@ impl LatencyRecorder {
 }
 
 /// The return value of message processors.
-///
-/// NOTE: In Python, this struct crosses a serialization boundary, and so this struct is somewhat
-/// sensitive to serialization speed. If there are additional things that should be returned from
-/// the Rust message processor that are not necessary in Python, it's probably best to duplicate
-/// this struct for Python as there it can be an internal type.
-#[derive(Clone, Debug, Default, Deserialize, Serialize, PartialEq)]
+#[derive(Clone, Debug, Default, PartialEq)]
 pub struct InsertBatch {
     pub rows: RowData,
     pub origin_timestamp: Option<DateTime<Utc>>,
     pub sentry_received_timestamp: Option<DateTime<Utc>>,
+    pub cogs_data: Option<CogsData>,
 }
 
 impl InsertBatch {
-    pub fn from_rows<T>(rows: impl IntoIterator<Item = T>) -> anyhow::Result<Self>
+    pub fn from_rows<T>(
+        rows: impl IntoIterator<Item = T>,
+        origin_timestamp: Option<DateTime<Utc>>,
```

Comment:
i think at some point we said that we wanted to get rid of `InsertBatch::from_rows` actually, but why make those changes in this PR?